cli/crates/server/src/bridge/listener/server.rs

```diff
@@ -0,0 +1,89 @@
+use super::super::consts::{DB_FILE, MODIFICATION_POLL_INTERVAL};
+use super::types::{EventRecord, Modification, StreamRecord};
+use crate::bridge::consts::{DB_URL_PREFIX, MODIFICATIONS_TABLE_NAME};
+use crate::{
+    errors::ServerError,
+    event::{wait_for_event, Event},
+};
+use chrono::Utc;
+use common::environment::Environment;
+use reqwest::Client;
+use sqlx::{query_as, Connection, SqliteConnection};
+use tokio::sync::broadcast::Sender;
+use tokio::time::sleep;
+use uuid::Uuid;
+
+async fn event_listener(worker_port: u16) -> Result<(), ServerError> {
+    let environment = Environment::get();
+    let db_file = environment.project_dot_grafbase_path.join(DB_FILE);
+
+    let db_url = match db_file.to_str() {
+        Some(db_file) => format!("{DB_URL_PREFIX}{db_file}"),
+        None => return Err(ServerError::ProjectPath),
+    };
+
+    let mut connection = SqliteConnection::connect(&db_url).await?;
+
+    let client = Client::new();
+
+    loop {
+        sleep(MODIFICATION_POLL_INTERVAL).await;
+
+        let delete_and_return_modifications = format!("DELETE FROM {MODIFICATIONS_TABLE_NAME} RETURNING *");
+
+        let modifications = query_as::<_, Modification>(&delete_and_return_modifications);
+
+        let results = match modifications.fetch_all(&mut connection).await {
+            Ok(results) => results,
+            // retry on the next interval if the DB is busy (due to a trigger writing an update)
+            Err(_) => {
+                // TODO: narrow this
+                continue;
+            }
+        };
+        if !results.is_empty() {
+            let dynamo_events = results
+                .iter()
+                .map(|result| EventRecord {
+                    aws_region: "us-east-1".to_owned(),
+                    change: StreamRecord {
+                        // as we poll 10 times per second and this is rounded to seconds,
+                        // using the current time is accurate enough.
+                        // using the record update time would be inaccurate for deletions
+                        approximate_creation_date_time: Utc::now().timestamp_millis() / 1000,
```

Comment:
OPT: It's ok, but couldn't just add the current timestamp in the SQLite trigger instead?

Reply:
We could probably just move this out of the loop for the same effect

Reply:
I'm not sure how accurate we need to be here

Reply:
But if SQLite has a nice way to generate a timestamp we could do that as well and add an extra column

Reply:
@Finistere done

Reply:
Ah missed the update! Thanks for the change, looks good to me!